src/common/logger.cpp

```diff
@@ -6,18 +6,55 @@
 
 #include "common/logger.hpp"
 
+#include <csignal>
 #include <cstdlib>
 
 #include "axom/slic.hpp"
 #include "mpi.h"
 
+namespace {
+/**
+ * The actual signal handler - must have this exact signature
+ * MPI is probably dead by this point, SLIC is definitely dead
+ */
+void signalHandler(int signal)
+{
+  std::cerr << "[SIGNAL]: Received signal " << signal << ", exiting" << std::endl;
+  serac::exitGracefully(true);
+}
+
+/**
+ * Registers the signalHandler function to handle various fatal
+ * signals
+ * @note The behavior of MPI when a signal is sent to the mpirun
+ * process is implementation-defined.  OpenMPI will send a SIGTERM
+ * (which can be caught) and then a SIGKILL (which cannot) a few
+ * seconds later.  PlatformMPI will propagate the signal to the
+ * individual processes.  Before propagating or sending a signal,
+ * MPI may or may not call MPI_Finalize/MPI_Abort.
+ */
+void registerSignals()
```

Comment:
There has been a desire to move this registerSignals logic into Axom's core library and add windows support or at least guard against it.

This goes with the need to handle SLIC's aborting for a more complete error handling functionality.

Reply:
Ok, we can put this branch on hold until equivalent functionality is provided by Axom.  I can move what I've got now onto an Axom branch.

> This goes with the need to handle SLIC's aborting for a more complete error handling functionality.

Just to confirm, does this refer to addressing the issues with `processAbort` and MPI in addition to https://github.com/LLNL/axom/issues/27 and https://github.com/LLNL/axom/issues/321?

Reply:
More of a move forward and when we think we have a hardened/working function we can move the logic into Axom to share with others.

As for the other, yea there is talk on the Axom team about reworking/improving how SLIC handles the aborting of parallel programs.  Not specifically that you fix the entire world.